libs/unity/library/Runtime/Scripts/Media/MediaLine.cs

```diff
@@ -0,0 +1,418 @@
+        /// or <see xref="Microsoft.MixedReality.WebRTC.MediaKind.Video"/>, respectively.
+        /// 
+        /// Internally the peer connection will automatically create and manage a media track to bridge the
+        /// media source with the transceiver.
+        ///
+        /// If this is non-<c>null</c> then the peer connection will negotiate sending some media, otherwise
+        /// it will signal the remote peer that it does not wish to send (receive-only or inactive).
+        ///
+        /// If <see cref="Transceiver"/> is valid, that is a first session negotiation has already been completed,
+        /// then changing this value raises a <see xref="WebRTC.PeerConnection.RenegotiationNeeded"/> event on the
+        /// peer connection of <see cref="Transceiver"/>.
+        /// </summary>
+        public MediaTrackSource Source
+        {
+            get { return _source; }
+            set
+            {
+                if (_source != value)
+                {
+                    _source = value;
+                    UpdateSenderOnSourceChanged();
+                    UpdateTransceiverDesiredDirection();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Media sender (track) bridging the <see cref="Source"/> with the underlying <see cref="Transceiver"/>.
+        /// 
+        /// The media track is automatically managed when <see cref="Source"/> changes or the transceiver direction
+        /// is renegotiated, as needed.
+        /// </summary>
+        public MediaSender Sender { get; private set; } = null;
+
+        /// <summary>
+        /// Backing field to serialize the <see cref="Receiver"/> property.
+        /// </summary>
+        /// <seealso cref="Receiver"/>
+        [SerializeField]
+        private MediaReceiver _receiver;
+
+        /// <summary>
+        /// Media receiver consuming the media received through the transceiver attached to this media line.
+        /// This must be an instance of a class derived from <see cref="AudioReceiver"/> or <see cref="VideoReceiver"/>
+        /// depending on whether <see cref="Kind"/> is <see xref="Microsoft.MixedReality.WebRTC.MediaKind.Audio"/>
+        /// or <see xref="Microsoft.MixedReality.WebRTC.MediaKind.Video"/>, respectively.
+        ///
+        /// If this is non-<c>null</c> then the peer connection will negotiate receiving some media, otherwise
+        /// it will signal the remote peer that it does not wish to receive (send-only or inactive).
+        ///
+        /// If <see cref="Transceiver"/> is valid, that is a first session negotiation has already been conducted,
+        /// then changing this value raises a <see xref="WebRTC.PeerConnection.RenegotiationNeeded"/> event on the
+        /// peer connection of <see cref="Transceiver"/>.
+        /// </summary>
+        public MediaReceiver Receiver
+        {
+            get { return _receiver; }
+            set
+            {
+                if (_receiver != value)
```

Comment:
This should check that the passed receiver has the correct type.

Reply:
Fixed; the setter now correctly checks the type of value.